smart-contract/test/gatewayNetwork.test.ts

```diff
@@ -53,21 +53,30 @@ describe('GatewayNetwork', () => {
     beforeEach('setup', async () => {
         [deployer, primaryAuthority, alice, bob, stableCoin, networkFeePayer] = await ethers.getSigners();
 
+        // Silence warnings from upgradable contracts with immutable variables
+        await upgrades.silenceWarnings();
+
         const gatewayNetworkFactory = await new GatewayNetwork__factory(deployer);
         const gatekeeperContractFactory = await new Gatekeeper__factory(deployer);
         const gatewayStakingFactory = await new GatewayStaking__factory(deployer);
         const dummyERC20Factory = await new DummyERC20__factory(deployer);
 
-        gatekeeperContract = await gatekeeperContractFactory.deploy();
+        gatekeeperContract = await upgrades.deployProxy(gatekeeperContractFactory, [deployer.address], { kind: 'uups', unsafeAllow: ['state-variable-immutable']}) as Gatekeeper;
```

Comment:
I'm not going to point out all unsafeAllow, but I'd go though them and would probably replace them by in-code annotations.